app/controllers/unattended_controller.rb

```diff
@@ -217,8 +217,14 @@ def update_ip
     logger.debug "Built notice from #{ip}, current host ip is #{@host.ip}, updating" if @host.ip != ip
 
     # @host has been changed even if the save fails, so we have to change it back
-    old_ip = @host.ip
-    @host.ip = old_ip unless @host.update({'ip' => ip})
+    address = IPAddr.new(ip)
+    if address.ipv4?
+      old_ip = @host.ip
+      @host.ip = old_ip unless @host.update({'ip' => ip})
+    elsif address.ipv6?
```

Comment:
Please filter out link local addresses
```suggestion
    elsif address.ipv6? && !address.link_local?
```

You perhaps you want to log a message if it's ignored, but I wouldn't feel too strongly about it.